src/mocks.ts

```diff
@@ -133,16 +76,168 @@ export const reduceAllMocksForScenario = (
+  >((result, mock) => {
+    const { url, operations } = mock;
+
+    const operationsByName: Record<string, Operation> = result[url.toString()]
+      ? result[url.toString()]
+      : {};
+
+    operations.forEach(operation => {
+      // Always take the latest operation
+      operationsByName[operation.operationName] = operation;
+    });
+
+    result[url.toString()] = operationsByName;
+    return result;
+  }, {});
+  const graphQlMocks = Object.entries(graphQlMocksByUrlAndOperations).map(
+    ([url, operationsByName]) => {
+      return {
+        method: 'GRAPHQL',
+        url: RegExp(url.replace(/^\/(.*)\/$/, '$1')),
+        operations: Object.values(operationsByName)
+      };
+    }
+  ) as GraphQLMock[];
+
+  return (httpMocks as any).concat(graphQlMocks);
 };
 
+/**
+ * Mocks the right HTTP method for a REST mock.
+ */
+function handleRestMock({
+  method,
+  url,
+  response,
+  responseCode = 200,
+  responseHeaders,
+  delay = 0
+}: HttpMock) {
+  const finalResponse = {
+    body: response,
+    status: responseCode,
+    headers: responseHeaders
+  };
+
+  switch (method) {
+    case 'GET':
+      FetchMock.get(url, () => addDelay(delay).then(() => finalResponse), {
+        overwriteRoutes: false
+      });
+      XHRMock.get(url, xhrMockDelay(finalResponse, delay));
+      break;
+    case 'POST':
+      FetchMock.post(url, () => addDelay(delay).then(() => finalResponse), {
+        overwriteRoutes: false
+      });
+      XHRMock.post(url, xhrMockDelay(finalResponse, delay));
+      break;
+    case 'PUT':
+      FetchMock.put(url, () => addDelay(delay).then(() => finalResponse), {
+        overwriteRoutes: false
+      });
+      XHRMock.put(url, xhrMockDelay(finalResponse, delay));
+      break;
+    case 'DELETE':
+      FetchMock.delete(url, () => addDelay(delay).then(() => finalResponse), {
+        overwriteRoutes: false
+      });
+      XHRMock.delete(url, xhrMockDelay(finalResponse, delay));
+      break;
+    default:
+      throw new Error(
+        `Unrecognised HTTP method ${method} - please check your mock configuration`
+      );
+  }
+}
+
+/**
+ * Mocks the right HTTP method for a GraphQL mock.
+ */
+function handleGraphQLMock({ url, operations }: GraphQLMock) {
+  const graphQLErrorResponse = { errors: [] };
+  const findMockGet = (u: string) => {
+    const parsedUrl = new URL(u);
+    const operationName = parsedUrl.searchParams.get('operationName');
+    return operations.find(o => o.operationName === operationName);
+  };
+  const findMockPost = postBody => {
+    const jsonBody =
+      typeof postBody === 'string' ? JSON.parse(postBody) : postBody;
+    return operations.find(
+      ({ operationName }) => operationName === jsonBody.operationName
+    );
+  };
+  const delayedResponse = (
+    delay: number | undefined,
+    response: object | string
+  ) => {
+    return addDelay(delay ? delay : 0).then(() => response);
+  };
+
+  FetchMock.get(url, u => {
+    const mock = findMockGet(u);
```

Comment:
Either here or in `findMockGet`, you'll want to check that the type is not a mutation.

Reply:
Good spot!